GliaWidgets/Sources/View/Chat/ChatView.swift

```diff
@@ -521,11 +523,21 @@ extension ChatView {
         isAnyEngagementTypeAvailable
             .filter { !$0 }
             .assign(to: &$isTopBannerExpanded)
-        isAnyEngagementTypeAvailable
-            .map { !$0 }
+
+        Publishers.CombineLatest(isAnyEngagementTypeAvailable, $isTopBannerAllowed)
+            .map { typesAvailable, isTopBannerAllowed in
+                if !isTopBannerAllowed {
```

Comment:
I think comment explaining the criteria for top banner to be hidden would be helpful.
```suggestion
                // If top banner is explicitly not allowed, then there's no need to 
                // evaluate available types, and we just early out hiding top banner.
                if !isTopBannerAllowed {
```

Reply:
Fixed c46bbbd9a62674243927de18305f4c09a895a80b